### ISSUE TITLE
ECONNRESET errors when uploading large numbers of files (Support Ticket #3695828)

### ISSUE THREAD

**Issue by bit-saver** [vimeo/vimeo.js#73]
When uploading a large number (triple digits) of video files we start to see numerous ECONNRESET errors that look like this:
`{ Error: read ECONNRESET at _errnoException (util.js:1022:11) at TLSWrap.onread (net.js:628:25) code: 'ECONNRESET', errno: 'ECONNRESET', syscall: 'read' }`

We don't have a lot of other information to provide surrounding this error other than that it only happens when we use the Vimeo SDK upload function. The function's error callback is not triggered.
We have ruled out our implementation as being the culprit as we can upload to Cloudflare Stream in exactly same manner (same TUS method of upload). When we upload to Cloudflare Stream we don't see any of these ECONNRESET errors. We don't want to use Cloudflare Stream for other reasons, but I'm just trying to substantiate the fact that it's related to the Vimeo SDK.

**Comment by erunion** [comment]
Hey there, when you upload to Cloudflare Stream, what version of `tus-js-client` are you using? I wonder if we just need to update our dependency from `^1.4.5` to `^1.5`.

**Comment by bit-saver** [comment]
We are indeed using `^1.5.1` of `tus-js-client`.

**Comment by erunion** [comment]
Ok cool. I'll see if we can upgrade what we use and see if that fixes your problems.

**Comment by bit-saver** [comment]
Sounds good, and thank you for the very quick response!

**Comment by erunion** [comment]
Tagged to 2.1.0. Let me know if that resolves your issues.

**Comment by bit-saver** [comment]
Unfortunately, this did not solve the problem.
I'm going to do some additional investigation to determine if it's truly the Vimeo SDK causing the problem.  I plan to switch from using the vimeo.upload function to a custom TUS implementation to see if that changes anything.

I wish I could catch the error to track down the trace leading up to it but none of the error callbacks get triggered.

**Comment by erunion** [comment]
Rats, okay. Let us know.

**Comment by bit-saver** [comment]
So I've isolated the problem to an error caused in downloading and **not** uploading.  I found that we weren't properly catching errors on the downloads which happen at the same time uploads are going on.  I'm not sure why we didn't see this happen as much when we used Cloudflare but the problem doesn't appear to Vimeo anymore. 

I'm sorry for the false alarm!  But thank you for trying to help me with it.
